Initial RSpec implementation

Shmekal opened the pull request:
Commit description:
- remove all non-rspec test-files(scripts)
- rename spec/basic_spec.rb to spec/register_spec.rb and move it to spec/features/
- most important: reorganize register_spec.rb and respective class
  note: not sure if split checks by register is worth doing at all as register of initial query affects nothing - register of Title and Description depends only on the particular search result and URL is always downcase

Notes:
Такой громоздкой коммит получился. Впоследующем лучше разбивать на логические части?

Shmekal committed:
- remove duplicate dependency in Gemfile
- remove all non-rspec test-files(scripts)
- rename spec/basic_spec.rb to spec/register_spec.rb and move it to spec/features/
- most important: reorganize register_spec.rb and respective class
  note: not sure if split checks by register is worth doing at all as register of initial query affects nothing - register of Title and Description depends only on the particular search result and URL is always downcase

nonkor commented on spec/features/register_spec.rb:
```diff
@@ -0,0 +1,55 @@
+
+  feature "Upcase input" do
+
+    # not sure if curly bracket syntax is acceptable
```
yep, acceptable

nonkor commented on spec/features/register_spec.rb:
```diff
@@ -0,0 +1,55 @@
+      end
+
+      scenario "URL contains inputted keyword Downcase" do
+        expect(@page.first_result_url).to include('monster')
```
why downcased version here?

Shmekal replied:
the thing is that url register is always downcased. knowing that we always expect it that way(downcase)

nonkor commented on spec/features/register_spec.rb:
```diff
@@ -0,0 +1,55 @@
+
+    feature "First result" do
+      scenario "Title contains inputted keyword Upcase" do
+        expect(@page.first_result_title).to include('Monster')
```
upcase?

Shmekal replied:
@nonkor , yes. the register in description and title doesn't correlate with the initial query(keyword) register. so as long as we check the hardcoded values this check expects particular register as we know what result will be shown. the test will become falling once search engine changes the first result which will include the downcased keyword. and this is the weak point I don't know how to deal with. maybe we should refuse doing register checks for the title and description sections? but probably there's already a conventional solution for such cases.

nonkor replied:
I would explain it by a call, much quicker.

nonkor commented on spec/features/register_spec.rb:
```diff
@@ -0,0 +1,55 @@
+      end
+
+      scenario "Description contains inputted keyword Upcase" do
+        expect(@page.first_result_description).to include('Monster')
```
upcase?

Shmekal replied:
@nonkor see the :41 comment

nonkor commented on lib/page.rb:
```diff
@@ -40,4 +35,34 @@ def parse_all_sites
+  def parse_first_result(destination)
+    locator =
+      case destination
+        when :title
```
good for now, but would be better to wrap them into page objects in future.

Shmekal replied:
@nonkor attr_accessor's, right?

nonkor replied:
no, I mean [page object pattern](https://habrahabr.ru/company/wapstart/blog/138674/). I would recommend to read that doc.

nonkor reviewed (changes requested):
Strange PR name, especially if know there is no gemfile touching :)
But in general - good PR, taka a look on my small issues.

nonkor commented:
About the commit: this seems ok for me. But yeah, makes sense to granulate them by logical parts.

Shmekal committed:
various changes:
- remove duplicate in Gemfile(finally)
- split spec into two by the case(register) principle

Shmekal commented:
@nonkor not sure how we go on. but here's the last commit description:
- remove duplicate in Gemfile(finally)
- split spec into two by the case(register) principle

Shmekal committed:
- undo previous 'spec split into two', namely combine two files into one.
- extend scenarios amount with cyryllic and multi-word keyword cases and revise their structure.
- introduce given, background and subject(not an alias but also implemented) capybara aliases.
- Gemfile.lock update due to bundler update process.

nonkor commented on spec/features/basic_search_spec.rb:
```diff
@@ -0,0 +1,110 @@
+        end
+
+      end # context downcase
+
```
You like empty lines so much. I recommend to use [ruby-style-guide](https://github.com/bbatsov/ruby-style-guide).

nonkor reviewed (approved):
I like it. Great job!
Next step: create page objects by to page object pattern. Good implementation of this pattern for capybara is [site_prism](https://github.com/natritmeyer/site_prism)

Shmekal committed:
remove empty lines between feature/context nestings
